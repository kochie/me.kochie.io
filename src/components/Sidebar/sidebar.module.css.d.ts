export const blur: string
export const canvas: string
export const bgImage: string
export const sidebar: string
export const sidebarHeader: string
export const avatar: string
export const quote: string
export const bg: string
export const jumbotron: string
export const container: string
