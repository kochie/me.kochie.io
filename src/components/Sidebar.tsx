import * as React from "react";

import Connections from './Connections'

import "../styles/Sidebar.scss";
import avatar from "../images/me.jpg";

export default class Sidebar extends React.Component {
  render() {
    const quote = "I build things, write the code for them, and run from the resulting explosion."
    return (
      <div className="sidebar">
        <div className="image">
          <div className="blur" />
        </div>
        <div className="sidebar-header">
          <div className="avatar">
            <img src={avatar} />
          </div>
          <div className="quote">{quote}</div>
          <Connections />
        </div>
      </div>
    );
  }
}
