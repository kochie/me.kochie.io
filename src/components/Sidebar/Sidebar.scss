@import "../../styles/constants";

.blur {
  background-color: black;
  height: 100%;
  opacity: 0.6;
}

.canvas {
  // transform: scale(1.1);
  height: 100%;
  width: 100%;
  position: absolute;
  top:0;
  left:0;
  z-index: 1;
  // filter: blur(2px);
  // background-size: cover;
  // background-image: url("./images/background.jpg");
  // perspective: 1px;
  // transform: translateZ(-1px) scale(2);

  @media only screen and (max-width: $mobile-width) and (max-height: $mobile-height) and (min-resolution: $mobile-resolution),
  only screen and (max-width: $mobile-height) and (max-height: $mobile-width) and (min-resolution: $mobile-resolution) {
    display: none
  }
}

.bg-image {
    // transform: scale(1.1);
    height: 100%;
    width: 100%;
    position: absolute;
    top:0;
    left:0;
    z-index: 1;
    // filter: blur(2px);
    // background-size: cover;
    // background-image: url("./images/background.jpg");
    // perspective: 1px;
    // transform: translateZ(-1px) scale(2);
    @media only screen and (max-width: $mobile-width) and (max-height: $mobile-height) and (min-resolution: $mobile-resolution),
    only screen and (max-width: $mobile-height) and (max-height: $mobile-width) and (min-resolution: $mobile-resolution) {
      background-image: url("./images/background-1.jpg");
      opacity: 0.5;
      background-color: black;
      background-position: center;
      background-size: cover;
      filter: blur(2px);
      transform: scale(1.1);
    }
}

.sidebar {
  position: fixed;
  // min-height: 100vh;
  height: 100vh;
  overflow: hidden;
  background-color: black;
  // overflow-y: auto;
  // perspective: 1px;
  width: 33vw;
  @media only screen and (max-width: $tablet-width) and (max-height: $tablet-height) and (min-resolution: $tablet-resolution),
  only screen and (max-width: $tablet-height) and (max-height: $tablet-width) and (min-resolution: $tablet-resolution) {
    position: absolute;
    top: 0px;
    width: 100vw;
    height: 100vh;
  }
  @media only screen and (max-width: $mobile-width) and (max-height: $mobile-height) and (min-resolution: $mobile-resolution),
  only screen and (max-width: $mobile-height) and (max-height: $mobile-width) and (min-resolution: $mobile-resolution) {
    position: absolute;
    top: 0px;
    width: 100vw;
    height: 100vh;
  }
}

.sidebar-header {
  position: relative;
  display: flex;
  top: 0px;
  left: 0px;
  z-index: 2;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  align-content: center;
  text-align: center;
  padding-top: 5vh;
  // width: 100%;
  // height: 100%;
  @media only screen and (max-width: $tablet-width) and (max-height: $tablet-height) and (min-resolution: $tablet-resolution),
    only screen and (max-width: $tablet-height) and (max-height: $tablet-width) and (min-resolution: $tablet-resolution) {
    justify-content: center;
    padding: 0px;
    height: inherit;
  }
}

.avatar {
  height: 150px;
  width: auto;
}

.avatar > img {
  border-radius: 50%;
  width: 150px;
  height: auto;
}

.quote {
  padding-top: 30px;
  padding-left: 15%;
  padding-right: 15%;
  color: wheat;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
}

.bg {
  // position: absolute;
  display: block;
  // width: 100%;
  z-index: 1;
  opacity: 1;
}

.jumbotron {
  // position: absolute;
  color: white;
  text-shadow: #444 0 1px 1px;
  // background: transparent;
  z-index: 2;
}

.container {
  // position: relative;
  // background-color: grey;
  z-index: -1;
}