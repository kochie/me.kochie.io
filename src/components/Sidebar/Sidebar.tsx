'use client'
import React, { useRef, useEffect, ReactElement, useCallback } from 'react'
import NextImage from 'next/image'

import Connections from '@/components/Connections/Connections'

import style from './sidebar.module.css'

import avatar from '@/images/City2Surf_Crop.png'
const background = '/images/background-2.webp'

const Sidebar = (): ReactElement => {
  const canvas = useRef<HTMLCanvasElement>(null)
  const image = useRef<HTMLImageElement>(null)
  let imageY = useRef(0)

  const parallax = useCallback((): void => {
    // console.log(document.getElementById('main').scrollTop)

    if (!image.current) return
    if (!canvas.current) return

    const main = document.getElementById('main')
    if (!main) return

    const ctx = canvas.current.getContext('2d')
    if (ctx === null) return
    const blurSize = 2 * window.devicePixelRatio

    const imgHeight = image.current.naturalHeight
    const imgWidth = image.current.naturalWidth

    const innerHeight = document.documentElement.clientHeight
    const innerWidth = canvas.current.clientWidth
    if (innerWidth === 0) return

    const docHeight = main.scrollHeight

    const scale = imgHeight / (innerHeight * 1 * window.devicePixelRatio)
    // const scale = 0.85

    const m = (imgHeight - innerHeight * scale) / (docHeight - innerHeight)
    const displacement = main.scrollTop - imageY.current
    imageY.current += displacement * 0.01
    const sy =
      m * Math.min(Math.max(imageY.current, 0), docHeight - innerHeight)
    ctx.restore()
    ctx.save()
    canvas.current.height = canvas.current.clientHeight
    canvas.current.width = canvas.current.clientWidth
    ctx.fillRect(0, 0, innerWidth, innerHeight)
    ctx.filter = `blur(${blurSize}px)`
    ctx.globalAlpha = 0.7
    const sx = (imgWidth - innerWidth * scale) * 0.5
    // if (sy === 0) return;
    ctx.drawImage(
      image.current,
      Math.round(sx),
      Math.round(sy),
      Math.round(scale * innerWidth),
      Math.round(scale * innerHeight),
      -blurSize,
      -blurSize,
      Math.round(innerWidth) + blurSize * 2,
      Math.round(innerHeight) + blurSize * 2
    )
    window.requestAnimationFrame(parallax)
  }, [])

  useEffect(() => {
    let num: number
    image.current = new Image()
    image.current.src = background
    image.current.onload = (): void => {
      num = window.requestAnimationFrame(parallax)
    }

    return () => {
      window.cancelAnimationFrame(num)
    }
  }, [parallax])

  const quote =
    'I build things, write the code for them, and run from the resulting explosion.'

  return (
    <nav
      className="relative top-0 w-full min-h-screen flex flex-col justify-center xl:h-screen xl:overflow-hidden xl:bg-black xl:w-1/3"
      role="complementary"
    >
      <div className="flex flex-col z-10 text-center items-center h-full mt-20">
        <div className="rounded-full w-40 h-40 overflow-hidden">
          <NextImage
            src={avatar}
            className='hover:scale-125 transform-gpu transition duration-500'
            alt="me"
            placeholder="blur"
            sizes="100vw"
            style={{
              width: '100%',
              height: 'auto',
            }}
          />
        </div>
        <div className={style.quote}>{quote}</div>
        <Connections />
      </div>
      <canvas
        className="absolute top-0 h-screen w-full invisible xl:visible"
        ref={canvas}
      />
      <div className="overflow-hidden h-screen w-full absolute xl:hidden">
        <div className="transform-gpu scale-110">
          <div className="bg-city-image md:bg-airplane-image h-screen w-screen bg-cover bg-center filter blur-sm top-0" />
        </div>
      </div>
    </nav>
  )
}

export default Sidebar
