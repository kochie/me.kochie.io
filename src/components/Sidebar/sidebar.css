.blur {
  background-color: black;
  height: 100%;
  opacity: 0.6;
}

.canvas {
  height: 100%;
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
}

.bg-image {
  height: 100%;
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
}

.sidebar {
  position: fixed;
  height: 100vh;
  overflow: hidden;
  background-color: black;
  width: 33vw;
}

.sidebar-header {
  position: relative;
  display: flex;
  top: 0px;
  left: 0px;
  z-index: 2;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  align-content: center;
  text-align: center;
  padding-top: 5vh;
}

.avatar {
  height: 150px;
  width: auto;
}

.avatar > img {
  border-radius: 50%;
  width: 150px;
  height: auto;
}

.quote {
  padding-top: 30px;
  padding-left: 15%;
  padding-right: 15%;
  color: wheat;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
}

.bg {
  display: block;
  z-index: 1;
  opacity: 1;
}

.jumbotron {
  color: white;
  text-shadow: #444 0 1px 1px;
  z-index: 2;
}

.container {
  z-index: -1;
}

@media (--mobile-size) {
  .canvas {
    display: none;
  }
  .bg-image {
    background-image: url("/images/background-1.jpeg");
    opacity: 0.5;
    background-color: black;
    background-position: center;
    background-size: cover;
    filter: blur(2px);
    transform: scale(1.1);
  }
}

@media (--tablet-size) {
  .sidebar {
    position: absolute;
    top: 0px;
    width: 100vw;
    height: 100vh;
  }

  .sidebar-header {
    justify-content: center;
    padding: 0px;
    height: inherit;
  }
}
