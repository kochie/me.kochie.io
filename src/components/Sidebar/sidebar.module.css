.blur {
  background-color: black;
  height: 100%;
  opacity: 0.6;
}

.canvas {
  height: 100%;
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
}

.bg-image {
  height: 100%;
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
}

.sidebar {
  position: fixed;
  height: 100vh;
  overflow: hidden;
  background-color: black;
  width: 33vw;
}

.sidebar-header {
  position: relative;
  display: flex;
  top: 0px;
  left: 0px;
  z-index: 2;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  align-content: center;
  text-align: center;
  padding-top: 5vh;
}

.avatar {
  height: 150px;
  width: auto;
}

.avatar > img {
  border-radius: 50%;
  width: 150px;
  height: auto;
}

.quote {
  padding-top: 30px;
  padding-left: 15%;
  padding-right: 15%;
  color: wheat;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
}

.bg {
  display: block;
  z-index: 1;
  opacity: 1;
}

.jumbotron {
  color: white;
  text-shadow: #444 0 1px 1px;
  z-index: 2;
}

.container {
  z-index: -1;
}

@media (--mobile-size) {
  .canvas {
    display: none;
  }
  .bg-image {
    background-image: url("../../assets/images/background-1.webp");
    opacity: 0.5;
    background-color: black;
    background-position: center;
    background-size: cover;
    filter: blur(2px);
    transform: scale(1.1);
  }
}

@media (--tablet-size) {
  .sidebar {
    position: relative;
    top: 0px;
    width: 100vw;
    height: fit-content;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  .sidebar-header {
    justify-content: center;
    padding: 30px 0px;
    height: inherit;
  }
}
