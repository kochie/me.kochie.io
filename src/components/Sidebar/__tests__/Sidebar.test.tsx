import * as React from "react";
import * as renderer from "react-test-renderer";

import Sidebar from "../Sidebar";

it("sidebar renders correctly", () => {
  const tree = renderer.create(<Sidebar />).toJSON();
  expect(tree).toMatchSnapshot();
});
