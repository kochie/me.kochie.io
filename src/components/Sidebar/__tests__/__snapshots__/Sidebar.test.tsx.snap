// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidebar renders correctly 1`] = `
<div
  className="sidebar"
>
  <div
    className="sidebarHeader"
  >
    <div
      className="avatar"
    >
      <img
        src="/images/me.jpeg"
      />
    </div>
    <div
      className="quote"
    >
      I build things, write the code for them, and run from the resulting explosion.
    </div>
    <div
      className="connections"
    >
      <div
        className="grow"
      >
        <a
          className="linkedin"
          href="https://linkedin.com/in/rkkochie"
        />
      </div>
      <div
        className="grow"
      >
        <a
          className="twitter"
          href="https://twitter.com/kochie"
        />
      </div>
      <div
        className="grow"
      >
        <a
          className="instagram"
          href="https://instagram.com/rkkochie"
        />
      </div>
      <div
        className="grow"
      >
        <a
          className="github"
          href="https://github.com/kochie"
        />
      </div>
      <div
        className="grow"
      >
        <a
          className="medium"
          href="https://medium.com/@kochie"
        />
      </div>
      <div
        className="grow"
      >
        <a
          className="gitlab"
          href="https://gitlab.com/kochie"
        />
      </div>
      <div
        className="grow"
      >
        <a
          className="mastodon"
          href="https://melb.social/@kochie"
        />
      </div>
    </div>
  </div>
  <canvas
    className="canvas"
  />
  <div
    className="bgImage"
  />
</div>
`;
